### PR TITLE
Add docs for nn.Module

### PR DESCRIPTION
**Pull request by EsdeathYZH**
这个PR补充了nn.Module的文档，参见https://github.com/Oneflow-Inc/oneflow/issues/7504

### CODE REVIEW

File: python/oneflow/nn/module.py [hunk 1]
```diff
@@ -254,47 +333,229 @@ def _named_members(self, get_members_fn, prefix="", recurse=True):
...
+
+            >>> import oneflow.nn as nn
+            >>> l1 = nn.Linear(2, 2)
+            >>> l2 = nn.Linear(2, 2)
+            >>> net = nn.Sequential(l1, l2)
+            >>> for idx, m in enumerate(net.children()): # doctest: +SKIP
+            ...     print(idx, '->', m) # doctest: +SKIP
+
+            0 -> Linear(in_features=2, out_features=2, bias=True)
+            1 -> Linear(in_features=2, out_features=2, bias=True)
```
Comment (doombeaker) [inline python/oneflow/nn/module.py:466]:
```suggestion
            >>> for idx, m in enumerate(net.children()):
            ...     print(idx, '->', m)
            0 -> Linear(in_features=2, out_features=2, bias=True)
            1 -> Linear(in_features=2, out_features=2, bias=True)
```

这里是真实代码及输出，是可以不用 SKIP 的吧。

File: python/oneflow/nn/module.py [hunk 2]
```diff
@@ -254,47 +333,229 @@ def _named_members(self, get_members_fn, prefix="", recurse=True):
...
+            >>> l = nn.Linear(2, 2)
+            >>> net = nn.Sequential(l, l)
+            >>> for idx, m in enumerate(net.modules()): # doctest: +SKIP
+            ...     print(idx, '->', m) # doctest: +SKIP
+
+            0 -> Sequential(
+              (0): Linear(in_features=2, out_features=2, bias=True)
+              (1): Linear(in_features=2, out_features=2, bias=True)
+            )
+            1 -> Linear(in_features=2, out_features=2, bias=True)
```
Comment (doombeaker) [inline python/oneflow/nn/module.py:520]: 同上，这里是真实的代码，可以不用 SKIP

File: python/oneflow/nn/module.py [hunk 3]
```diff
@@ -475,6 +796,37 @@ def load(module, prefix=""):
...
+                returned in the state dict are detached from autograd. If it's
+                set to ``True``, detaching is not performed. Default: ``False``
+
+        Returns:
+            dict:
+                a dictionary containing a whole state of the module
+
+        Example::
+
+            >>> module.state_dict().keys() # doctest: +SKIP
```
Comment (doombeaker) [inline python/oneflow/nn/module.py:826]: 这里我感觉是不是改成一个真实的例子，比如 Linear，会更好

File: python/oneflow/nn/module.py [hunk 4]
```diff
@@ -52,6 +52,38 @@ def _addindent(s_, numSpaces):
...
+    Submodules assigned in this way will be registered, and will have their
+    parameters converted too when you call :meth:`to`, etc.
+
+    .. note::
+        As per the example above, an ``__init__()`` call to the parent class
+        must be made before assignment on the child.
+
+    :ivar training: Boolean represents whether this module is in training or
+                    evaluation mode.
+    :vartype training: bool
```
[the remark this answers is not in the document]
Comment (EsdeathYZH) [reply python/oneflow/nn/module.py:84]: 效果如下：
image

File: python/oneflow/nn/module.py (hunk 3 above)
Comment (EsdeathYZH) [reply python/oneflow/nn/module.py:826]: done

File: python/oneflow/nn/module.py [hunk 5]
```diff
@@ -308,16 +569,51 @@ def named_modules(self, memo: Optional[Set["Module"]] = None, prefix: str = ""):
...
+        eval()
+        
+        Sets the module in evaluation mode.
+
+        This has any effect only on certain modules. See documentations of
+        particular modules for details of their behaviors in training/evaluation
+        mode, if they are affected, e.g. :class:`Dropout`, :class:`BatchNorm`,
+        etc.
+
+        This is equivalent with :meth:`self.train(False) <oneflow.nn.Module.train>`.
```
[the remark this answers is not in the document]
Comment (EsdeathYZH) [reply python/oneflow/nn/module.py:605]: 效果如下，BatchNorm上面没有挂上链接：
image

File: python/oneflow/nn/module.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (EsdeathYZH) [reply python/oneflow/nn/module.py:88]: 看了下pytorch的，它那里BatchNorm也没链接

File: python/oneflow/nn/module.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (EsdeathYZH) [reply python/oneflow/nn/module.py:606]: 看了下pytorch的，它那里BatchNorm也没链接

File: python/oneflow/nn/module.py [hunk 6]
```diff
@@ -554,12 +945,91 @@ def can_use_assign_copy(tensor, tensor_applied):
...
+            tensor([[1., 1.],
+                    [1., 1.]], dtype=oneflow.float32, requires_grad=True)
+            Sequential(
+              (0): Linear(in_features=2, out_features=2, bias=True)
+              (1): Linear(in_features=2, out_features=2, bias=True)
+            )
+            Sequential(
+              (0): Linear(in_features=2, out_features=2, bias=True)
+              (1): Linear(in_features=2, out_features=2, bias=True)
+            )
```
Comment (kevinJhz) [inline python/oneflow/nn/module.py:985]: 这里Sequential()只会被打印1次，可以改一下

File: python/oneflow/nn/module.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (kevinJhz) [reply python/oneflow/nn/module.py:606]: oneflow没有BatchNorm这个class（torch也没有），可以改成BatchNorm1d、BatchNorm2d？

[the remark this answers is not in the document]
Comment (EsdeathYZH) [reply python/oneflow/nn/module.py:606]: 我觉得可以

**Review by kevinJhz — APPROVED** [review]

**Review by doombeaker — APPROVED** [review]
